stix2/core.py

```diff
@@ -286,11 +319,12 @@ def _register_observable_extension(
                 ext_type,
             )
     else:  # 2.1+
-        if not re.match(SCO21_EXT_REGEX, ext_type):
+        if not re.match(TYPE_21_REGEX, ext_type):
```

Comment:
This is for custom SCO extensions: shouldn't the extension name be matched to `EXT_21_REGEX`?

Reply:
Good catch - fixed